modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/Commons.java

```diff
@@ -142,8 +142,9 @@ public final class Commons {
                     .withExpand(false)
                     .withHintStrategyTable(
                             HintStrategyTable.builder()
-                                    .hintStrategy("DISABLE_RULE", (hint, rel) -> true)
-                                    .hintStrategy("EXPAND_DISTINCT_AGG", (hint, rel) -> rel instanceof Aggregate)
+                                    .hintStrategy(Hints.ENFORCE_JOIN_ORDER.name(), (hint, rel) -> true)
```

Comment:
why did we want to hold predicate rule configuration as lambdas ? i.e. : rel instanceof Aggregate ? 
check for example : org.apache.calcite.rel.hint.HintPredicates seems it`s more clear and seems it`s @lowka is talking about.
Additionally why new rule is propageted for all rels ? Not for : **rel instanceof Join** ?

Reply:
good catch, thanks. Fixed. But it's not the same which talk @lowka about